src/main/java/techreborn/packets/ServerboundPackets.java

```diff
@@ -171,6 +174,29 @@ public static void init() {
 				}
 			});
 		}));
+
+		NetworkManager.registerServerBoundHandler(LAUNCH_SPEED, ((server, player, handler, buf, responseSender) -> {
+			BlockPos pos = buf.readBlockPos();
+			int buttonAmount = buf.readInt();
+
+			server.execute(() -> {
+				BlockEntity blockEntity = player.world.getBlockEntity(pos);
+				if (blockEntity instanceof LaunchpadBlockEntity) {
+					((LaunchpadBlockEntity) blockEntity).handleGuiInputFromClient(buttonAmount);
+				}
+			});
+		}));
+
+		NetworkManager.registerServerBoundHandler(new Identifier("reborncore", "jump"), (server, player, handler, buf, responseSender) -> {
+			BlockPos pos = buf.readBlockPos();
+
+			server.execute(() -> {
+				BlockEntity blockEntity = player.getWorld().getBlockEntity(pos.down());
+				if (blockEntity instanceof ElevatorBlockEntity elevator) {
+					elevator.teleportUp(player.getWorld(), pos.down(), player);
```

Comment:
You should check that the player is within a few blocks of the evevator, as right now someone could easily send this packet from anywhere in the world and get teleported.

Reply:
doesnt the code require that the block below the player is an elevator?

Reply:
Doesnt look like the code does. You cannot trust any packet data coming from the client.

Reply:
Ah, I see my mistake now. But I would wait with fixing until the "event" is properly implemented